languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.metadata.*
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import groovy.transform.*
+import com.ibm.jzos.ZFile
+import com.ibm.dbb.build.report.*
+import com.ibm.dbb.build.report.records.*
+
+
+// define script properties
+@Field BuildProperties props = BuildProperties.getInstance()
+@Field def buildUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BuildUtilities.groovy"))
+@Field def impactUtils= loadScript(new File("${props.zAppBuildDir}/utilities/ImpactUtilities.groovy"))
+@Field def bindUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BindUtilities.groovy"))
+@Field def resolverUtils = loadScript(new File("${props.zAppBuildDir}/utilities/ResolverUtilities.groovy"))
+
+println("** Building files mapped to ${this.class.getName()}.groovy script")
+
+// verify required build properties
+buildUtils.assertBuildProperties(props.cc_requiredBuildProperties)
+
+// create language datasets
+def langQualifier = "cc"
+buildUtils.createLanguageDatasets(langQualifier)
+
+// sort the build list based on build file rank if provided
+List<String> sortedList = buildUtils.sortBuildList(argMap.buildList, 'cc_fileBuildRank')
+
+if (buildListContainsTests(sortedList)) {
+    langQualifier = "cc_test"
+    buildUtils.createLanguageDatasets(langQualifier)
+}
+
+// iterate through build list
+sortedList.each { buildFile ->
+    println "*** Building file $buildFile"
+
+    // Check if this a testcase
+    isZUnitTestCase = (props.getFileProperty('cc_testcase', buildFile).equals('true')) ? true : false
+
+    // configure dependency resolution and create logical file
+    String dependencySearch = props.getFileProperty('cc_dependencySearch', buildFile)
+    def dependencyResolver = resolverUtils.createSearchPathDependencyResolver(dependencySearch)
+
+    // copy build file and dependency files to data sets
+    if(isZUnitTestCase){
+        buildUtils.copySourceFiles(buildFile, props.cc_testcase_srcPDS, null, null, null)
+    }else{
+        buildUtils.copySourceFiles(buildFile, props.cc_srcPDS, 'cc_dependenciesDatasetMapping', props.cc_dependenciesAlternativeLibraryNameMapping, dependencyResolver)
+    }
+
+
+    // Get logical file
+    LogicalFile logicalFile = resolverUtils.createLogicalFile(dependencyResolver, buildFile)
+
+    // create mvs commands
+    String member = CopyToPDS.createMemberName(buildFile)
+    File logFile = new File( props.userBuild ? "${props.buildOutDir}/${member}.log" : "${props.buildOutDir}/${member}.c.log")
+    if (logFile.exists())
+        logFile.delete()
+    MVSExec compile = createCompileCommand(buildFile, logicalFile, member, logFile)
+    MVSExec linkEdit = createLinkEditCommand(buildFile, logicalFile, member, logFile)
```

Comment:
It's better to only construct the linkEdit MVSExec if the build file `needsLinking`. Let's move the construction of the linkEdit step into the condition, please.